### PR TITLE
Rotate back across the globe when points cross the IDL

### PR DESCRIPTION
**Pull request by lasalvavida**
Fix for #3874.

### CODE REVIEW

**Comment by lasalvavida** [comment]
Corrected output:

capture

**Comment by hpinkos** [comment]
I don't think that's quite right @lasalvavida.  One of the blue rectangles should be rotated.  The result should look like the yellow rectangles.

**Comment by hpinkos** [comment]
Can you add a unit test for this?

**Comment by lasalvavida** [comment]
As discussed offline, this is the expected output:

```javascript
var western2 = createWesternRectangle(0.005);
```

The blue rectangle is only rotated a small amount.

**Comment by lasalvavida** [comment]
Adding a test now. Just to be clear, the crux of this issue is that the rectangle has its northwest corner on the opposite of the IDL from its center. So, when we do the following: 

```javascript
 nwCartesian = Cartesian3.subtract(nwCartesian, centerCartesian, nwCartesian);
```

Instead of getting the very small difference between them, we subtract a negative value, producing a large offset which blows up small rotations to the point where it causes a translation.

File: Source/Core/RectangleGeometryLibrary.js [hunk 1]
```diff
@@ -118,6 +118,10 @@ define([
             granYSin = granularityY * sinRotation;
             granXSin = granularityX * sinRotation;
 
+            if (center.longitude < nwCorner.longitude) {
+                center.longitude += 2 * Math.PI;
```
Comment (hpinkos) [inline Source/Core/RectangleGeometryLibrary.js]: do `CesiumMath.TWO_PI`

**Comment by hpinkos** [comment]
Looks good to me @lasalvavida!  I just had that one comment.

**Comment by lasalvavida** [comment]
Test added!

**Comment by hpinkos** [comment]
Great, thanks @lasalvavida!

**Comment by pjcozzi** [comment]
@hpinkos please update CHANGES.md in master.

**Pull request merged**